src/main/java/gregtech/api/items/metaitem/MusicDiscStats.java

```diff
@@ -0,0 +1,48 @@
+package gregtech.api.items.metaitem;
+
+import gregtech.api.items.metaitem.stats.IItemBehaviour;
+import gregtech.api.items.metaitem.stats.IMusicDisc;
+import net.minecraft.block.BlockJukebox;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraft.stats.StatList;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+public class MusicDiscStats implements IMusicDisc, IItemBehaviour {
+
+    private final SoundEvent sound;
+
+    public MusicDiscStats(SoundEvent sound) {
+        this.sound = sound;
+    }
+
+    @SideOnly(Side.CLIENT)
+    @Override
+    public SoundEvent getSound() {
+        return sound;
+    }
+
+    @Override
+    public ActionResult<ItemStack> onItemUse(EntityPlayer player, World world, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+        IBlockState iblockstate = world.getBlockState(pos);
+        ItemStack itemStack = player.getHeldItem(hand);
+        if (iblockstate.getBlock() == Blocks.JUKEBOX && !(Boolean)iblockstate.getValue(BlockJukebox.HAS_RECORD)) {
+            if (!world.isRemote) {
+                ((BlockJukebox)Blocks.JUKEBOX).insertRecord(world, pos, iblockstate, itemStack);
+                world.playEvent(1010, pos, itemStack.getItemDamage());
```

Comment:
how about a new type defined? no need to hack the getItemByID. just chech the type at head of the `playEvent`. Avoid breaking switch, and better compatibility.

Reply:
Not a bad idea, but doing this would require a coremod section for the JukeBox as well, to cancel the SoundEvent of the record playing, since it is hardcoded to send a null on the 1010 event code when a record is popped out